test/e2e/e2e_test.go

```diff
@@ -87,6 +89,83 @@ var _ = Describe("E2E", func() {
 				// TODO Check that the expected CRDs, deployments, services, etc. are present
 			})
 		})
+		Context("Mixerless Telemetry Stats Filter Test", func() {
+			// var resourcesFile string
+			var (
+				istio        v1beta1.Istio
+				filterBefore *bool
+				err          error
+				majorMinor   string
+			)
+			const timeout = 30 * time.Second
+			const interval = 1 * time.Second
+
+			JustBeforeEach(func() {
+				// get the istio object created in the OUTER JustBeforeEach
+				log.Info("Namespace: ", "Namespace", instance.Namespace)
+				log.Info("Name: ", "Name", instance.Name)
+				filterBefore, err = GetMixerlessTelemetryStatus(&istio, instance.Namespace, instance.Name)
+				Expect(err).NotTo(HaveOccurred())
+				log.Info("JustBeforeEach: ", "filterBefore", filterBefore)
+				version := istio.Spec.Version // get first 2 digits
+				versionParts := strings.SplitN(string(version), ".", 3)
+				majorMinor = fmt.Sprintf("%s.%s", versionParts[0], versionParts[1])
+				log.Info("Istio Version: ", "Version", majorMinor)
+
+			})
+
+			AfterEach(func() {
+				var filterNow *bool
+				filterNow, err = GetMixerlessTelemetryStatus(&istio, instance.Namespace, instance.Name)
+				Expect(err).NotTo(HaveOccurred())
+				// Only change back if filterBefore != filterNow
+				// complicated by the ability of the *bool to be nil
+				if (filterNow == nil && filterBefore != nil) || (filterNow != nil && filterBefore == nil) || (
+					filterNow != nil && filterBefore != nil && *filterNow != *filterBefore) {
+					log.Info("AfterEach: Restore", "filterBefore", filterBefore, "filterNow", filterNow)
+					err = SetMixerlessTelemetryState(&istio, filterBefore)
+					Expect(err).NotTo(HaveOccurred())
+				}
```

Comment:
Is this really necessary? I mean `SetMixerlessTelemetryState(&istio, filterBefore)` should be enough, it doesn't matter what the current state is.

But even that's more than necessary: the istio resource will be deleted right after the test (look for `istioTestEnv.Close()`) and then the GC deletes all referencing resources, including the envoyfilters. What needs to be cleaned up, are the resources created "by hand", for example the meshgateway resource in one of the other test cases. There are no such resources in this test, so I think this `AfterEach` can be removed.

And I think that means the `filterBefore` variable is unnecessary, too.

Reply:
Removed the tracking of the initial state and the cleanup